signac/contrib/project.py

```diff
@@ -255,18 +252,6 @@ def workspace(self):
         """
         return self._workspace
 
-    @property
-    def id(self):
-        """Get the project identifier.
```

Comment:
Personally I'd like to avoid the redundancy, especially because depending on how users make use of the property it could lead to subtle bugs if we change the meaning of the property without removing it. For instance, it wouldn't surprise me if a user was making use of the id to customize filenames or something. Obviously not the intended use case, but anything is possible. That said, I agree that removing it could be quite disruptive. I'd be fine with following the precedent we set for `doc_filter` and mark this as a deprecation to be removed in signac 3.0, redefining it as an alias of `root_directory` during 2.0. Thoughts?

Reply:
I favor making a decision in 2.0 rather than deferring to 3.0, but which approach to take (removing id vs. redefining it) is not clear to me. I’d like others’ thoughts here.

Reply:
If the uniquely identifying feature is the path (similar to a SQlite database), then that should be it, no redundancy. In so far I would advocate for removing `id` with 2.0.

Reply:
I also think we should just remove `id` in 2.0. We could always post in our #general channel and see if there are any users that really need `id` but it might be better to start the notice that it will be removed now and see if anyone asks about it, we could in the message even direct people to file and issue if they need it and then we can discuss it.

Reply:
Taking these comments as a vote to move forward with removal. We can gauge community opinions like Mike suggested, and we'll also introduce a deprecation warning in the 1.x line in #644 that should warn users loudly.